src/axom/spin/policy/LinearBVH.hpp

```diff
@@ -58,7 +58,10 @@ class LinearBVHTraverser
   {
     auto traversePref = [](const BoxType& l, const BoxType& r, const PointType& p) {
       double sqDistL = primal::squared_distance(p, l.getCentroid());
-      double sqDistR = primal::squared_distance(p, r.getCentroid());
+      // If the right bbox is not valid, return max. Otherwise, the invalid right
+      // bbox might actually win when we should ignore it.
+      double sqDistR = r.isValid() ? primal::squared_distance(p, r.getCentroid())
```

Comment:
I think the centroid returned 0, which seemed ok to me. It looks like the Doxygen code would not indicate a isValid restriction.

```
/*!
   * \brief Returns the centroid (midpoint) of the bounding box.
   * \return Point at the bounding box centroid.
   */
```

Reply:
Thanks for checking. That seems reasonable, but we should probably add a note in the doxygen that `centroids` really only make sense for valid/non-empty bounding boxes.

Reply:
I don't think `getCentroid` returns zero for an invalid box.  It returns some form of infinity.  `squared_distance` seems to treat it as (0,0) though.  This could be confusing for users.